### PR TITLE
fix(parser): fetch json keys regardless of cases

### PR DESCRIPTION
**Pull request by idx0-dev**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
I have removed case-converting in simd-json.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by lmatz** [comment]
Does this intend to fix #8894,
How about adding some test cases for it?

Rest LGTM!

File: src/connector/src/parser/common.rs [hunk 1]
```diff
@@ -25,6 +27,22 @@ use simd_json::value::StaticNode;
...
+) -> Option<&'b BorrowedValue<'a>> {
+    let obj = v.as_object()?;
+    if obj.contains_key(key.as_ref()) {
+        return obj.get(key.as_ref());
+    }
+    let lower_key = key.to_ascii_lowercase();
+    for (k, v) in obj {
+        if k.to_ascii_lowercase() == lower_key {
+            return Some(v);
+        }
```
Comment (tabVersion) [inline src/connector/src/parser/common.rs:42]: `to_ascii_lowercase` requires extra allocation, may try `eq_ignore_ascii_case`

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Comment by lmatz** [comment]
Does `name_in_lower_case` become totally useless and thus we can remove it?

**Comment by idx0-dev** [comment]
> 



> Does `name_in_lower_case` become totally useless and thus we can remove it?
I don't know. I don't have any context about why does it exist.

**Comment by lmatz** [comment]
I mean 
https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/parser/maxwell/simd_json_parser.rs#L73-L92
maxwell also uses it

if this also needs to be removed, then there will no usage of this `name_in_lower_case` I suppose 🤔

**Comment by idx0-dev** [comment]
> I mean https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/parser/maxwell/simd_json_parser.rs#L73-L92 maxwell also uses it
> 
> if this also needs to be removed, then there will no usage of this `name_in_lower_case` I suppose 🤔

Oh, yes. I will replace it (maxwell) tomorrow.

**Comment by lmatz** [comment]
No worries, let's just merge this: https://github.com/risingwavelabs/risingwave/pull/9106

**Pull request merged**